core/src/main/scala-3/medeia/generic/GenericProductDecoder.scala

```diff
@@ -33,30 +35,29 @@ object GenericProductDecoder {
       labelling: Labelling[A],
       options: GenericDerivationOptions[A]
   ): EitherNec[BsonDecoderError, A] = {
-    val (acc, result) = inst.unfold[Accumulator](Accumulator(labelling.elemLabels, Chain.empty))(
+    val (acc, result) = inst.unfold[Accumulator](Accumulator(labelling.elemLabels, None))(
       [t] =>
-        (acc: Accumulator, rt: BsonDecoder[t]) => {
-          val label = options.transformKeys(acc.labels.head)
+        (acc: Accumulator, rt: BsonDecoder[t]) =>
+          val label =
+            acc.labels.headOption
+              .map(options.transformKeys)
+              .getOrElse(throw new RuntimeException("Bug in derivation logic, accumulator has no labels!"))
```

Comment:
Added an explicit exception instead of the `.head` call.  I was thinking if this is actually a valid case?  Are there any `empty` products?

Reply:
Yes, case objects are empty products

Reply:
Thanks!